cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/PreferencesHttpHandler.java

```diff
@@ -216,6 +250,17 @@ public void getProgramPrefs(HttpRequest request, HttpResponder responder,
     }
   }
 
+  @Path("/namespaces/{namespace-id}/apps/{application-id}/{program-type}/{program-id}/preferences/metadata")
```

Comment:
this type of path is normally used to fetch a specific object in the 'preferences' collection. In other words, it would normally be used to get the value of the preference named 'metadata'. 'metadata' is also an overloaded term. Metadata in CDAP means something very specific and should not be re-used in an inconsistent way. 

Ideally we would add a new field to the existing response, though that doesn't seem possible here because the response is just a map.

Are we sure we need this sequence id thing? How is it being used?

Reply:
Can we have path name like: .../preference_metadata?

From what I can tell, we do need this seq id for validation purpose. Client currently validates that this seq id < most recent seq id, so this can guard against bugs.

Reply:
I see. I don't think we want to use 'metadata', since it means something specific elsewhere in CDAP. We don't have underscores in other places, perhaps something like preferenceDetail. 

Also, I think the response should include both the sequence id as well as the actual preferences associated with that id. Otherwise you have another race where you can't be sure that the preferences call matches the result of this call. This also applies to the PreferencesTable, we shouldn't have separate methods for getting the sequence number and the actual preferences.